core/vm/evm.go

```diff
@@ -224,7 +243,8 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 	}
 
 	if isPrecompile {
-		ret, gas, err = RunPrecompiledContract(p, input, gas)
+		// TODO(yevhenii): deduct gas?
```

Comment:
Worth adding test cases for this, I think the cosmos-sdk side of things will handle tracking and deducting actual gas (with a possible assist from statedb @drklee3 ?) but having unit and integration tests around what happens gas wise when a block with a failed precompile tx is executed are a useful foundation for testing the feature now and deeper as needed going forward

Reply:
fyi: PR was moved to https://github.com/Kava-Labs/go-ethereum/pull/10

gas deduction changed a bit in a new PR, also I added some tests related to gas calculations